orttraining/orttraining/test/training_api/test_runner.cc

```diff
@@ -0,0 +1,323 @@
+    if (params.gradient_accumulation_steps < 1) {
+      return Status(ONNXRUNTIME, INVALID_ARGUMENT,
+                    "Invalid gradient_accumulation_steps parameter: should be >= 1");
+    }
+
+    params.train_data_dir = ToPathString(flags["train_data_dir"].as<std::string>());
+    params.test_data_dir = ToPathString(flags["test_data_dir"].as<std::string>());
+    params.output_dir = ToPathString(flags["output_dir"].as<std::string>());
+    if (params.output_dir.empty()) {
+      printf("No output directory specified. Trained model files will not be saved.\n");
+    }
+
+    session_options.use_deterministic_compute = flags["use_deterministic_compute"].as<bool>();
+
+    ort_params.log_severity = static_cast<logging::Severity>(flags["ort_log_severity"].as<int>());
+    ORT_RETURN_IF_NOT(
+        logging::Severity::kVERBOSE <= ort_params.log_severity && ort_params.log_severity <= logging::Severity::kFATAL,
+        "Log severity must be in the range [", static_cast<int>(logging::Severity::kVERBOSE),
+        ", ", static_cast<int>(logging::Severity::kFATAL), "].");
+    ort_params.vlog_level = flags["ort_vlog_level"].as<int>();
+  } catch (const exception& e) {
+    const std::string msg = "Failed to parse the command line arguments";
+    cerr << msg << ": " << e.what() << "\n"
+         << options.help() << "\n";
+    return Status(ONNXRUNTIME, INVALID_ARGUMENT, msg);
+  }
+
+  return Status::OK();
+}
+
+template <typename T>
+static void CreateInputOrtValue(gsl::span<const int64_t> dims,
+                                const std::vector<T>& value,
+                                OrtValue* p_ortvalue,
+                                AllocatorPtr alloc = nullptr) {
+  static CPUExecutionProviderInfo info;
+  static CPUExecutionProvider cpu_provider(info);
+  static AllocatorPtr cpu_allocator = cpu_provider.GetAllocator(0, OrtMemTypeDefault);
+
+  TensorShape shape(dims);
+  assert(shape.Size() == static_cast<int64_t>(value.size()));
+  auto element_type = DataTypeImpl::GetType<T>();
+  auto allocator = alloc ? alloc : cpu_allocator;
+  auto p_tensor = std::make_unique<Tensor>(element_type, shape, allocator);
+
+  if (value.size() > 0) {
+    memcpy(p_tensor->MutableDataRaw(), value.data(), p_tensor->SizeInBytes());
+  }
+
+  p_ortvalue->Init(p_tensor.release(),
+                   DataTypeImpl::GetType<Tensor>(),
+                   DataTypeImpl::GetType<Tensor>()->GetDeleteFunc());
+}
+
+std::vector<std::vector<OrtValue>> CreateTestDataLoader(AllocatorPtr alloc = nullptr) {
+  OrtValue input, positions;
+  // hard code batch size = 4.
+  CreateInputOrtValue(std::array<int64_t, 4>{4}, std::vector<int64_t>{1, 2, 3, 4}, &input, alloc = alloc);
+  CreateInputOrtValue(std::array<int64_t, 4>{4}, std::vector<int64_t>{1, 2, 3, 3}, &positions, alloc = alloc);
+  return std::vector<std::vector<OrtValue>>(20, std::vector<OrtValue>{input, positions});
+}
+
+float GetLossValue(const Tensor& loss_tensor) {
```

Comment:
maybe this can be a more generic function. Also we don't plan on having fp16 support so might not be needed